packages/flame/lib/src/events/messages/position_event.dart

```diff
@@ -8,13 +9,15 @@ import 'package:meta/meta.dart';
 /// This class includes properties that describe the position where the event
 /// has occurred.
 abstract class PositionEvent extends Event {
-  PositionEvent({required this.canvasPosition, required this.devicePosition});
+  PositionEvent(this._game, {required this.devicePosition});
+  final Game _game;
 
   /// Event position in the coordinate space of the game widget, i.e. relative
   /// to the game canvas.
   ///
   /// This could be considered the Flame-level global position.
-  final Vector2 canvasPosition;
+  late final Vector2 canvasPosition =
+      _game.convertGlobalToLocalCoordinate(devicePosition);
```

Comment:
Hmm, if this is used a lot it's going to be quite expensive when used in events, but I guess it already is.
Something that we should look into in the future though, how to not create so many new Vector2 objects when doing this.